vertx-web/src/test/java/io/vertx/ext/web/handler/RerouteTest.java

```diff
@@ -59,4 +59,36 @@ public void testRerouteMethod() throws Exception {
 
     testRequest(HttpMethod.GET, "/me", 200, "OK", "POST");
   }
+
+  @Test
+  public void testRerouteWithBody() throws Exception {
+    router.route("/test/*").handler(BodyHandler.create());
+    router.route("/test/v1").handler(ctx -> {
+      ctx.reroute("/test/v2");
+    });
+    router.route("/test/v2").handler(ctx -> {
+      ctx.response().end();
+    });
+
+    testRequest(HttpMethod.POST, "/test/v1", req -> {
+      req.setChunked(true);
+      req.write("Test HTTP Body");
+    }, 200, "OK", null);
+  }
+
+//  @Test
```

Comment:
Contents of commented Test are identical to the test above. You can delete the commented Test, I guess..